File: temp/Baekjoon_17143.java
package temp;
// bufferedReader가 scanner보다 빠르지만, type을 바꿔줘야하는 번거로움 존재
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.FileInputStream;
import java.util.StringTokenizer;
import java.util.Arrays;

class Baekjoon_17143 {
    public static void main(String args[]) throws Exception {
        System.setIn(new FileInputStream("temp/baekjoon_17143.txt"));
        BufferedReader bReader = new BufferedReader(new InputStreamReader(System.in));
        
        String[] input = bReader.readLine().split(" ");
        int R = Integer.parseInt(input[0]);
        int C = Integer.parseInt(input[1]);
        int M = Integer.parseInt(input[2]);

        int[][] arr = new int[M][5];
        for (int i=0; i<M; ++i) {
            StringTokenizer st = new StringTokenizer(bReader.readLine()," ");
            for (int j=0; j<5; ++j){
                arr[i][j] = Integer.parseInt(st.nextToken());
            }   
        } 
        long answer = 0;
        

        System.out.println(Arrays.toString(arr[0]));
      
    }
}

File: temp/baekjoon_17143.py
# 2마리 이상의 상어일 경우 가장 큰 상어가 다잡아먹음 -> 크기가 같은 경우는? 상어의 방향을 정해야함
import sys, os
BASE_DIR = os.path.splitext(os.path.realpath(__file__))[0] +  '.txt'
sys.stdin = open(BASE_DIR)
input = sys.stdin.readline

R,C,M = map(int,input().split())

arr = [ list(map(int,input().split())) for _ in range(M)]
answer = 0

mat = [[-1]*C for _ in range(R)]

for i in range(M):
    mat[arr[i][0]-1][arr[i][1]-1] = i

for p in range(C):
    for r in range(R):
        shark = mat[r][p]
        if shark != -1:
            answer+=arr[shark][4]
            mat[r][p] = -1
            arr[shark] = None
            break
    
    for j in range(M):
        if arr[j]!=None:
            mat[arr[j][0]-1][arr[j][1]-1] = -1
            s = arr[j][2]
            d = arr[j][3]
            while s>0:
                s-=1
                if d == 1:
                    arr[j][0]-=1
                    if arr[j][0] == 0:
                        d = 2
                        arr[j][0] += 2
                elif d == 2:
                    arr[j][0]+=1
                    if arr[j][0] == R+1:
                        d = 1
                        arr[j][0] -= 2
                elif d == 3:
                    arr[j][1]+=1
                    if arr[j][1] == C+1:
                        d = 4
                        arr[j][1] -= 2
                elif d == 4:
                    arr[j][1]-=1
                    if arr[j][1] == 0:
                        d = 3
                        arr[j][1] += 2
            arr[j][3] = d
            mx = mat[arr[j][0]-1][arr[j][1]-1]
            if mx == -1:
                mat[arr[j][0]-1][arr[j][1]-1] = j
            elif mx < j:
                if arr[mx][4] < arr[j][4]:
                    arr[mx] = None
                    mat[arr[j][0]-1][arr[j][1]-1] = j
                else:
                    arr[j] = None
print(answer)
    